builder/azure/arm/step_deploy_template.go

```diff
@@ -112,13 +126,14 @@ func (s *StepDeployTemplate) Cleanup(state multistep.StateBag) {
 		}
 		if !state.Get(constants.ArmKeepOSDisk).(bool) {
 			ui.Say(fmt.Sprintf(" Deleting -> %s : '%s'", imageType, imageName))
-			err = s.deleteDisk(context.TODO(), imageName, resourceGroupName, isManagedDisk)
+			err = s.deleteDisk(ctx, imageName, resourceGroupName, (isManagedDisk || isSIGImage), subscriptionId, armStorageAccountName)
```

Comment:
This is the change that fixes https://github.com/hashicorp/packer-plugin-azure/issues/315, since before SIG images would use the VHD deletion path